specs/src/specs/block_proposer.md

```diff
@@ -15,11 +15,9 @@ There are two restrictions on the original data's square size:
 
 With these restrictions in mind, the block proposer performs the following actions:
 
-1. Collect as many transactions and messages from the mempool as possible, such that the total number of shares is at most [`AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`](./consensus.md#constants).
+1. Collect as many transactions and blobs from the mempool as possible, such that the total number of shares is at most [`AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`](./consensus.md#constants).
 1. Compute the smallest square size that is a power of 2 that can fit the number of shares.
-1. Attempt to [lay out the collected transactions and messages](#laying-out-transactions-and-messages) in the current square.
-    1. If the square is too small to fit all transactions and messages (which may happen [due to needing to insert padding between messages](../rationale/message_block_layout.md)) and the square size is smaller than [`AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`](./consensus.md#constants), double the size of the square and repeat the above step.
+1. Attempt to lay out the collected transactions and blobs in the current square.
+    1. If the square is too small to fit all transactions and blobs (which may happen [due to needing to insert padding between blobs](../rationale/data_square_layout.md)) and the square size is smaller than [`AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`](./consensus.md#constants), double the size of the square and repeat the above step.
```

Comment:
You may want to add that if the square size is currently the `AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`, than we result to dropping transactions and blobs until the data fits within the square

Reply:
Resolved by 7c8b7becec46687e12b41a7e59d81312fa8e6452